internal/pkg/routes/mfx/mfx.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mfx
+
+import (
+	"fmt"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/export/internal/pkg/routes"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/senml"
+	"github.com/nats-io/nats.go"
+)
+
+const (
+	// JSON represents SenML in JSON format content type.
+	JSON = "application/senml+json"
+
+	// CBOR represents SenML in CBOR format content type.
+	CBOR = "application/senml+cbor"
+)
+
+var formats = map[string]senml.Format{
+	JSON: senml.JSON,
+	CBOR: senml.CBOR,
+}
+
+type mfxRoute struct {
+	route routes.Route
+}
+
+func NewRoute(n, m, s string, mqtt mqtt.Client, l logger.Logger) routes.RouteIF {
+	return &mfxRoute{
+		route: routes.Route{
```

Comment:
I think that you can use `NewRoute` from `routes` package here. Also, I think about giving separate logger to the `mfgRoute`. The only reason that `Route` is public here is because of the logger (if you remove `Mqtt` usage), so if you add logger to the `mfxRoute`, then you'll be able to just use Route interface.

Reply:
Route is public so that mfxRoute can "inherit"
```

	NatsTopic string
	MqttTopic string
	Subtopic  string
```
which are common fields, and there will be possible other fields ( direction nats-nats, nats-mqtt, mqtt-nats, mqtt-mqtt )
I need NewRoute so that I have right implementation of RouteIF interfejs
Some other implementation of RouteIF can have different `Consume` that would perfrom different processing